distribution/tools/plugin-cli/src/main/java/org/opensearch/plugins/InstallPluginCommand.java

```diff
@@ -864,16 +878,11 @@ private PluginInfo installPlugin(Terminal terminal, boolean isBatch, Path tmpRoo
         }
         PluginSecurity.confirmPolicyExceptions(terminal, permissions, isBatch);
 
-        final Path destination = env.pluginsFile().resolve(info.getName());
+        String folderName = (info.getFolderName() == null || info.getFolderName().isEmpty()) ? info.getName() : info.getFolderName();
```

Comment:
Should be a method on `info`, e.g. `info.getTargetFolderName()` for example, and reused in a couple of places where we try to do this condition.

Reply:
Added a method `getTargetFolderName()` for `PluginInfo`.

Reply:
This is not a must have, but you could probably do what I am suggesting above, set the value of `folderName` in the constructor and then you don't even need `getTargetFolderName` - `folderName` would have the correct value all the time.